.github/workflows/ci.yml

```diff
@@ -233,6 +233,34 @@ jobs:
           # Might be set to warn for debugging purposes. Warning, log file will be huge.
           RADIXDLT_LOG_LEVEL: error
         run: ./gradlew clean runTargetedIntegrationTests --info --refresh-dependencies --parallel
+  mesh-api-test-suite:
+    name: Run Mesh API tests
+    runs-on: selfhosted-ubuntu-22.04-16-cores
+    steps:
+      - uses: RDXWorks-actions/checkout@main
+        with:
+          # Shallow clones should be disabled for a better relevancy of analysis
+          fetch-depth: 0
+      - name: Setup environment
+        uses: ./.github/actions/setup-env
+      - name: Cache Gradle packages
+        uses: RDXWorks-actions/cache@main
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+          restore-keys: ${{ runner.os }}-gradle
+      - name: Install mesh-cli
+        run: curl -sSfL https://raw.githubusercontent.com/coinbase/mesh-cli/master/scripts/install.sh | sh -s
+      - name: Generate keygen
+        run: ./gradlew keygen:run --args='-k ../core/node.ks -p ""'
+      - name: Build Node
+        run: ./gradlew build
+      - name: Run Node in the background
+        run: ./gradlew :core:run --info &
```

Comment:
To save the keygen step, you could just use `RADIXDLT_NODE_KEY=AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAY=` here